Content.Server/GameObjects/Components/Weapon/Melee/MeleeWeaponComponent.cs

```diff
@@ -55,6 +55,11 @@ public class MeleeWeaponComponent : Component, IAttack
         [ViewVariables(VVAccess.ReadWrite)]
         public bool ClickAttackEffect { get; set; }
 
+        /// <summary>
+        ///     Returns the set of entities hit by a melee attack.
+        /// </summary>
+        public Action<IEnumerable<IEntity>> OnHittingEntities;
```

Comment:
Should be a component / eventbus message instead. Main reason is you need to unsubscribe from this but there's no guarantee what order it happens when components are deleted.

Should also be a list so you can easily get the count rather than each potential subscriber getting it individually.